builtin/providers/azure/resource_azure_hosted_service.go

```diff
@@ -0,0 +1,190 @@
+package azure
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/management"
+	"github.com/Azure/azure-sdk-for-go/management/hostedservice"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// resourceAzureHostedService returns the schema.Resource associated to an
+// Azure hosted service.
+func resourceAzureHostedService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAzureHostedServiceCreate,
+		Read:   resourceAzureHostedServiceRead,
+		Update: resourceAzureHostedServiceUpdate,
+		Delete: resourceAzureHostedServiceDelete,
+
+		SchemaVersion: 1,
+
+		Schema: map[string]*schema.Schema{
+			"service_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: parameterDescriptions["service_name"],
+			},
+			"location": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: parameterDescriptions["location"],
+			},
+			"url": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ephemeral_contents": &schema.Schema{
+				Type:     schema.TypeBool,
+				Required: true,
+				DefaultFunc: func() (interface{}, error) {
+					return false, nil
+				},
```

Comment:
The docstrings in the Schema say that if Required is set you cannot use Default; but using DefaultFunc is permitted.
Will have to ask for a follow-up on this before doing any changes.